test/vector_search/vector_store_client_test.cc

```diff
@@ -882,3 +892,29 @@ SEASTAR_TEST_CASE(vector_store_client_load_balancing) {
                 co_await s2->stop();
             }));
 }
+
+SEASTAR_TEST_CASE(vector_search_metrics_test) {
+    
+    auto cfg = cql_test_config();
+    cfg.db_config->vector_store_primary_uri.set("http://good.authority.here:6080");
+    co_await do_with_cql_env(
+            [](cql_test_env& env) -> future<> {
+                auto as = abort_source();
+                auto schema = co_await create_test_table(env, "ks", "test");
+                auto result = co_await env.execute_cql("CREATE CUSTOM INDEX idx ON ks.test (embedding) USING 'vector_index'");
+                result.get()->throw_if_exception();
+                auto& vs = env.local_qp().vector_store_client();
+
+                vs.start_background_tasks();
+                auto all_metrics = seastar::metrics::impl::get_values();
+                auto dns = get_metrics_value("vector_store_dns_refreshes", all_metrics)->i();
+                dns++;
+                vector_store_client_tester::trigger_dns_resolver(vs);
+                BOOST_CHECK(co_await repeat_until(seconds(1), [&dns]() -> future<bool> {
+                    auto all_metrics = seastar::metrics::impl::get_values();
+                    auto new_dns = get_metrics_value("vector_store_dns_refreshes", all_metrics)->i();
```

Comment:
Nitpick: I would have added a short coawait seastar::sleep() somewhere in the loop, otherwise this is needlessly a tight loop wasting 100% of the CPU (and not being able to make progress anyway until the repeat_until decides to preempt when its quota is done).